test/es-module/test-esm-detect-ambiguous.mjs

```diff
@@ -294,7 +306,11 @@ describe('Module syntax detection', { concurrency: !process.env.TEST_PARALLEL },
         'const fs = require("node:fs"); await Promise.resolve();',
       ]);
 
-      match(stderr, /ReferenceError: require is not defined in ES module scope/);
+      match(
+        stderr,
+        /SyntaxError: Top-level await is not supported in CommonJS modules\. To use top-level await, add "type": "module" to your package\.json or rename the file to use the \.mjs extension\. Alternatively, wrap the await expression in an async function\./
+      );
```

Comment:
This test is titled “throws on undefined `require` when top-level `await` triggers ESM parsing”. That’s not happening anymore per your assertion here.

I think erroring about the top-level `await` instead of the undefined `require` _is_ a better approach, but then the test title needs updating.